agent/engine/execcmd/manager.go

```diff
@@ -115,3 +116,12 @@ func IsExecEnabledContainer(container *apicontainer.Container) bool {
 	_, ok := container.GetManagedAgentByName(ExecuteCommandAgentName)
 	return ok
 }
+
+// HasExecAgentInitFailed returns if exec feature has been disabled for a container
+// because of exec agent initialization failure. This method checks if the exec agent
+// is stopped and its ID is an empty string. The ID can only be an empty string
+// if the initialization of the exec agent fails. It is only after successful exec
+// agent initialization, we assign a uuid to the ID
+func HasExecAgentInitFailed(ma apicontainer.ManagedAgent) bool {
+	return ma.ID == "" && ma.Status == apicontainerstatus.ManagedAgentStopped
```

Comment:
This is going to be tricky if the assumptions change in the future. I'd prefer to add a boolean flag to `ManagedAgentState` and set it explicitly if initialization of exec agent fails